src/main/java/org/apache/commons/lang3/time/FastDateParser.java

```diff
@@ -535,6 +542,7 @@ boolean isNumber() {
 
         @Override
         boolean parse(final FastDateParser parser, final Calendar calendar, final String source, final ParsePosition pos, final int maxWidth) {
+            @SuppressWarnings("argument.type.incompatible") // cannot annotate pos.getIndex() as @LTEqLengthOf("source"), but pos.getIndex() return the current parse position, which has to be @LTEqLengthOf("source")
```

Comment:
Why can't it be annotated that way?

Reply:
Because in the stub file, no string argument is there to annotate it accordingly. 
But pos.getIndex() returns the current parse position, i.e, index at which parsing will begin.

Reply:
Sorry, you are right. This is fine as-is.